contracts/staking/src/execute.rs

```diff
@@ -744,15 +759,16 @@ pub fn update_config_from_migrate(
 ) -> ContractResult<Response> {
     let mut config: Config = CONFIG.load(deps.storage)?;
    
-    // update oracle contract address v2
-    if msg.oracle_contract_address_v2.is_some() {
-        let oracle_contract_address_v2 = msg.oracle_contract_address_v2.unwrap();
-        let address = validate_address(&oracle_contract_address_v2, "osmo")?;
-        config.oracle_contract_address_v2 = Some(address);
+    // update oracle contract address v3
+    if msg.oracle_address.is_some() {
+        let oracle_address = msg.oracle_address.unwrap();
+        let address = validate_address(&oracle_address, "osmo")?;
+        config.oracle_address = Some(address);
     }
+
     CONFIG.save(deps.storage, &config)?;
 
-    Ok(Response::new().add_attribute("action", "update_oracle_contract_address_v2"))
+    Ok(Response::new().add_attribute("action", "update_oracle_address"))
```

Comment:
From my perspective, we can ignore this. 
The suggested code will make the contract panic if we receive an invalid address, while with the original code, we return an error to the user. 
If we want to make this more idiomatic Rust, I would prefer to remove the `unwrap` call from inside the `map` and use `transpose` on the returned `Option<Result<Addr>>` so that we can return the error to the user instead of panicking. However, this is more of a coding style decision that we should all agree on.